src/type_id.h

```diff
@@ -197,4 +197,10 @@ struct construction;
 using construction_id = int_id<construction>;
 using construction_str_id = string_id<construction>;
 
+struct generic_event_type;
+using generic_event_type_id = string_id<generic_event_type>;
+
+struct generic_requirement_type;
+using generic_requirement_type_id = string_id<generic_requirement_type>;
+
```

Comment:
These typedefs are intended to be in alphabetical order.  I see there are already a few that are not, but please add yours in order.